### PR TITLE
executor: remove Next function for prepared.go

### PR DESCRIPTION
**Pull request by yangwenmai**
remove Next function for PrepareExec && ExecuteExec && DeallocateExec.

### CODE REVIEW

File: executor/prepared.go [hunk 1]
```diff
@@ -236,11 +225,6 @@ type DeallocateExec struct {
 	Name string
 }
 
-// Next implements the Executor Next interface.
-func (e *DeallocateExec) Next(ctx context.Context) (Row, error) {
-	return nil, errors.Trace(e.run(ctx))
```
Comment (XuHuaiyu) [inline executor/prepared.go]: Put the code of e.run here.

Comment (yangwenmai) [reply executor/prepared.go]: why?

Comment (zz-jason) [reply executor/prepared.go]: `e.run` is only called by `NextChunk`, we no longer need it.

Comment (XuHuaiyu) [reply executor/prepared.go]: The logic of this code is very simple,
no need to be extracted since Next is removed.

File: executor/prepared.go [hunk 2]
```diff
@@ -93,11 +93,6 @@ func NewPrepareExec(ctx sessionctx.Context, is infoschema.InfoSchema, sqlTxt str
 	}
 }
 
-// Next implements the Executor Next interface.
-func (e *PrepareExec) Next(ctx context.Context) (Row, error) {
-	return nil, errors.Trace(e.DoPrepare())
```
Comment (zz-jason) [inline executor/prepared.go]: function `e.DoPrepare` can also be removed.

File: executor/prepared.go (hunk 1 above)
Comment (zz-jason) [reply executor/prepared.go]: `e.run` can also be removed.

Comment (zz-jason) [reply executor/prepared.go]: @yangwenmai plz address the comment of @XuHuaiyu

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

**Pull request merged**